packages/jest-util/lib/jasmine-pit.js

```diff
@@ -5,82 +5,55 @@
-    throw new Error(`pit(...): tests must return a promise`);
-  }
+function isPromise(obj) {
+  return obj && typeof obj.then === 'function';
 }
 
-function wrapForJasmine1(env, promiseFn) {
-  return () => {
-    let error = null;
-    let isFinished = false;
-    const currentSpec = env.currentSpec;
-    currentSpec.runs(() => {
-      try {
-        const promise = promiseFn.call(currentSpec);
-        assertPromise(promise);
-        promise
-          .then(() => isFinished = true)
-          .catch(err => {
-            error = err;
-            isFinished = true;
-          });
-      } catch (e) {
-        error = e;
-        isFinished = true;
-      }
-    });
-    currentSpec.waitsFor(() => isFinished);
-    currentSpec.runs(() => {
-      if (error) {
-        throw error;
-      }
-    });
-  };
-}
+// return a wrapping function: `env.fit = makePromiseIt(env, 'fit')`
+function makePromiseIt(env, fnName) {
+  const originalIt = env[fnName];
 
-function wrapForJasmine2(promiseFn) {
-  return done => {
-    const promise = promiseFn();
-    const fail = done.fail ? done.fail : err => { throw err; };
+  if (typeof originalIt !== 'function') {
+    throw new Error(`trying to wrap env.${fnName} that is not a function`);
+  }
 
-    try {
-      assertPromise(promise);
-    } catch (err) {
-      fail(err);
+  return function(specName, fn) {
+    const isAsync = !!fn.length; // `done` was passed
+
+    if (isAsync) {
+      return originalIt.call(env, specName, fn); // jasmine will handle it
+    } else {
+      // we make *all* tests async and run `done` right away if they
+      // didn't return a promise.
+      return originalIt.call(env, specName, function(done) {
+        const returnValue = fn.apply(this, arguments);
+
+        if (isPromise(returnValue)) {
+          returnValue.then(done).catch(done.fail)
+        } else if (returnValue === undefined) {
+          done();
+        } else {
+          done.fail(new Error('`it` must return either promise or nothing'));
+        }
+      });
     }
-
-    promise
-      .then(res => done(res))
-      .catch(err => fail(err));
   };
 }
 
 function install(global) {
   const jasmine = global.jasmine;
   const env = jasmine.getEnv();
-  const isJasmine1 = jasmine.version_ && jasmine.version_.major === 1;
-  const wrapFn = isJasmine1 ? wrapForJasmine1.bind(null, env) : wrapForJasmine2;
-
-  global.pit = function pit(specName, promiseFn) {
-    return env.it(specName, wrapFn(promiseFn));
-  };
-  global.xpit = function xpit(specName, promiseFn) {
-    return env.xit(specName, wrapFn(promiseFn));
-  };
-
-  if (isJasmine1) {
-    global.pit.only = function pitOnly(specName, promiseFn) {
-      return env.it.only(specName, wrapFn(promiseFn));
-    };
-  } else {
-    global.fpit = function fpit(desc, promiseFn) {
-      return env.fit(desc, wrapFn(promiseFn));
-    };
-  }
+  // env.xit will not run anyways so we don't have to worry about it
+  env.it = makePromiseIt(env, 'it');
```

Comment:
let's make this so we it looks like this: `env.it = promisifyIt(env.it, env);`? Passing function names as strings is dangerous.

Also, can we do:

`env.pit = env.it`? We shouldn't break `pit` just yet – we can deprecate it and provide a codemod so people can switch and we'll remove it later in a major version. Here is what I recommend (in case this diff doesn't impact perf too much):

* Alias `it` to `pit`.
* Codemod everything on Facebook's www to `it`.
* Add a warning to `pit`.